### DISCUSSION TITLE
ERROR: Async Storage has been extracted from react-native core and will be removed in a future release

### DISCUSSION THREAD

**Discussion by yubliwarokkaid** [react-native-async-storage/async-storage#555] [General]
Why can an error appear like this in the terminal when running React Native?

ERROR: Async Storage has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/async-storage' instead of 'react-native'. See https://github.com/react-native-community/react-native-async-storage

**Comment by tido64** [comment]
This was part of the "Lean Core" effort. Essentially, modules that weren't "core" to React Native functionalities, were extracted to separate repos. When they were pulled out, they were replaced with error messages like the one you encountered. You can read more about it in this issue: https://github.com/facebook/react-native/issues/23313

**Reply by LeonieHeiduk** [reply]
Hey! Sorry if this is a dumb question - but I am still getting the error, even though I am importing from react-native-async-storage and cleaned the cache. Some say you should just suppress the warning - is that the solution? or am I overlooking something? Thanks in advance!

**Reply by tido64** [reply]
If you are importing `react-native-async-storage` and you're still getting the warning, then something else is using the `react-native` one. I would go through my dependencies to see which one it is and upgrade that package. If that's not possible, I would align with the package to avoid duplicates in your code and ignore the warning. You should also open an issue in their repo to get them to migrate. We don't have a timeline for when Fb decides to remove AsyncStorage from react-native core.
